### PR TITLE
react-recap-issue3

### PR DESCRIPTION
**Pull request by SylvieDoumet**
Issue 3 - please review and confirm merging option

### CODE REVIEW

File: src/Components/Color/Color.jsx [hunk 1]
```diff
@@ -12,6 +12,10 @@ export default function Color({ color }) {
       <h3 className="color-card-headline">{color.hex}</h3>
       <h4>{color.role}</h4>
       <p>contrast: {color.contrastText}</p>
+
+      {/* Issue 3: Delete Color - Added a delete button to each color card and passed it a 'onDelete' handler */}
+
+      <button onClick={onDelete}>Delete</button>
```
Comment (JessicaLoers) [inline src/Components/Color/Color.jsx]: Please take a look into the dev tools.  You will see the following error message.
```
ReferenceError: onDelete is not defined
```

**What could be the reason for this?**

File: src/App.jsx [hunk 2]
```diff
@@ -2,29 +2,79 @@ import { initialColors } from "./lib/colors";
...
+// Issue #2 Add Color - add function to handle color adding
   const handleAddColor = (newColor) => {
     setColors([{ ...newColor, id: nanoid() }, ...colors]);
   };
 
+// Issue #3 Delete Color - add function to handle color deletion 
+const handleDeleteColor = (id) => {
+  setColors(colors.filter(color => color.id !== id);
+  setConfirmDeleteId(null);
+};
```
Comment (JessicaLoers) [inline src/App.jsx]: Why did you choose to use arrow functions for the handlers? 

You update the state `confirmDeleteId ` from `null` to `null`? Think about what exactly the goal is.
 Have you checked the functionality of your app?

File: src/App.jsx [hunk 3]
```diff
@@ -2,29 +2,79 @@ import { initialColors } from "./lib/colors";
...
+{/* Issue #3 Delete Color  - add function to display message when no colors are left */}
+{colors.length === 0 ? (
+<p>No colors present. Please add new colors.</p>
+) : (
+
+    colors.map((color) => (
+        <Color 
+        key={color.id} 
+        color={color} 
+        onDelete={() => handleConfirmDelete(color.id)}
```
Comment (JessicaLoers) [inline src/App.jsx]: Shouldn't this function delete the colour? Why are using the `handleConfirmDelete` ?
Have you checked the functionality in your app?

File: src/App.jsx [hunk 4]
```diff
@@ -2,29 +2,79 @@ import { initialColors } from "./lib/colors";
...
+// Issue #3 Delete Color - add function to handle color deletion 
+const handleDeleteColor = (id) => {
+  setColors(colors.filter(color => color.id !== id);
+  setConfirmDeleteId(null);
+};
+
+// Issue #3 Delete Color - add function to handle confirmation state 
+
+const handleConfirmDelete = (id) => {
+  setConfirmDeleteId(null);
```
Comment (JessicaLoers) [inline src/App.jsx]: Why did you choose to use arrow functions for the handlers? 

Is the function parameter necessary?

You update the state `confirmDeleteId ` from `null` to `null`? Think about what exactly the goal is.
 Have you checked the functionality of your app?

File: src/App.jsx [hunk 5]
```diff
@@ -2,29 +2,79 @@ import { initialColors } from "./lib/colors";
...
+};
+
+const handleCancelDelete = (id) => {
+  setConfirmDeleteId(null);
+};
```
Comment (JessicaLoers) [inline src/App.jsx]: Why did you choose to use arrow functions for the handlers?

Is the function parameter necessary?

You update the state confirmDeleteId  from `null` to `null`? Think about what exactly the goal is.
Have you checked the functionality of your app?

File: src/App.jsx [hunk 6]
```diff
@@ -2,29 +2,79 @@ import { initialColors } from "./lib/colors";
...
+ 
+ {/* Issue #3 Delete Color - add a delete button to each color card */}
+ 
+{confirmDeleteId && (
+  <div className="confirmation-dialog">
+  <p className="color-card-highlight">Are you sure you want to delete?</p>
+  <button onClick={() => handleDeleteColor(setConfirmDeleteId)}>Yes</button>
+<button onClick={handleCancelDelete}>No</button>
+  </div>
+)}
```
Comment (JessicaLoers) [inline src/App.jsx]: If you want to work with the `setConfirmDeleteId` state, it's important that you specify the corresponding ID.

Here you call the `handleDeleteColor` function and pass it a state setter function - that's wild. 🎢. 
However, your delete function expects an ID

**Review by JessicaLoers — CHANGES_REQUESTED** [review]
**Hi @SylvieDoumet** 

⚠️ The feature to delete a color isn't working. Therefore, this issue is not yet resolved. 
Please thoroughly test your application before submitting your code for review. Ensure that all acceptance criteria are met.

When programming, it isn't only advisable but also very sensible to use the Dev Tools. This would have helped you catch the reference error. 😉

**Questions:**
- Why did you choose to use arrow functions for the handlers?
- Why are the component imports incorrect? Did you check this? Is the `ColorInput` Component folder in the correct location? 

**Additional Notes:**
Please work with the project board and adhere to the workflow:
- Move the issue you are currently working on to the "In Progress" column.
- Close issues that you have already completed.
- Please deploy to Vercel.

**Comment by SylvieDoumet** [comment]
Ich habe jetzt alle Anmerkungen bearbeitet - habe allerdings in der ColorForm.jsx immer noch Importprobleme und sehe nicht wo der Fehler liegt 
Ich habe jetzt mal die neuste Version gepushed - sollte ich erneut einen pull request stellen?

**Pull request merged**